src/test/java/org/apache/commons/compress/archivers/tar/TarArchiveEntryTest.java

```diff
@@ -306,6 +308,164 @@ public void getOrderedSparseHeadersRejectsStructsPointingBeyondOutputEntry() thr
         te.getOrderedSparseHeaders();
     }
 
+    @Test
+    public void shouldParseTimePaxHeadersAndNotCountAsExtraPaxHeaders() {
+        TarArchiveEntry entry = createEntryForTimeTests();
+        assertEquals("extra header count", 0, entry.getExtraPaxHeaders().size());
+        assertNull("size", entry.getExtraPaxHeader("size"));
+        assertNull("mtime", entry.getExtraPaxHeader("mtime"));
+        assertNull("atime", entry.getExtraPaxHeader("atime"));
+        assertNull("ctime", entry.getExtraPaxHeader("ctime"));
+        assertNull("birthtime", entry.getExtraPaxHeader("LIBARCHIVE.creationtime"));
+        assertEquals("size", entry.getSize(), 1);
+        assertEquals("mtime", FileTime.from(Instant.parse("2022-03-14T01:25:03.599853900Z")), entry.getLastModifiedTime());
+        assertEquals("atime", FileTime.from(Instant.parse("2022-03-14T01:31:00.706927200Z")), entry.getLastAccessTime());
+        assertEquals("ctime", FileTime.from(Instant.parse("2022-03-14T01:28:59.700505300Z")), entry.getStatusChangeTime());
+        assertEquals("birthtime", FileTime.from(Instant.parse("2022-03-14T01:29:00.723509000Z")), entry.getCreationTime());
+    }
+
+    @Test
+    public void shouldNotWriteTimePaxHeadersByDefault() throws IOException {
+        TarArchiveEntry entry = createEntryForTimeTests();
+
+        final ByteArrayOutputStream bos = new ByteArrayOutputStream();
```

Comment:
Use try-with-resources. I know this is will not cause a real resource leak on a thrown exception this avoids compiler warnings and it is a best practice we want to exemplify.

Reply:
Done

Reply:
Btw, I can maybe add try-with-resouces on this ByteArrayOutputStream, but the TarArchiveOutputStream's close methd already closes the BAOS as well. If I do add the BAOS to the try-with-resources, I'll have to create a byte array to save its internal byte array after closing the TarArchiveOutputStream, meaning a nested try-with-resources. Let me know what you prefer, as both are ok examples because of how TarArchiveOutputStream works.

Reply:
Leave it as is then. I don't think you'd need need nested try-w-res since you can have >1 statement in a try. But it's a detail we do not need to deal with as a requirement for this PR to make it. I'll leave it up to you.

Reply:
I'd think that would be an antipattern in a way, because I'd have to get the data from the BAOS before both it and the TarArchiveOutputStream are closed if both are on the same try-with-resources, which is generally not a good idea with some OutputStream implementations.

I'd rather have them nested to ensure the TarArchiveOutputStream is closed before getting the data from the BAOS as that's the safest thing to do in most scenarios.